### PR TITLE
[MM-41668] Remove "add support email" admin advisor notifications

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary
As requested in the ticket [MM-41668](https://mattermost.atlassian.net/browse/MM-41668
), we're dropping `admin-advisor`'s notification to the admins about having a missing support email.

**Note:** I went "all-in" in my removal, dropping absolutely everything related to this feature that is not used anywhere else - I like not having unused code whenever possible. Please let me know if there's any desire to keep certain methods, like `notifyAdminsOfWarnMetricStatus` that exists to send a notification to the admins but is not used anywhere else.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-41668

#### Release Note
```release-note
Removed `admin-advisor` bot ability to notify admins about missing support email
```

### CODE REVIEW

File: app/admin_advisor.go [hunk 1]
```diff
@@ -150,128 +146,6 @@ func (a *App) getWarnMetricStatusAndDisplayTextsForId(warnMetricId string, T i18
 	return nil, nil
 }
 
-func (a *App) notifyAdminsOfWarnMetricStatus(c *request.Context, warnMetricId string, isE0Edition bool) *model.AppError {
```
Comment (JulienTant) [inline app/admin_advisor.go:153]: Removed as a side-effect: this is not used anywhere else

File: app/server_test.go [hunk 2]
```diff
@@ -706,91 +705,3 @@ func TestSentry(t *testing.T) {
 		}
 	})
 }
-
-func TestAdminAdvisor(t *testing.T) {
```
Comment (JulienTant) [inline app/server_test.go:710]: This test only covers the support email notifications feature

**Review by marianunez — APPROVED** [review]

**Comment by JulienTant** [comment]
the QA commit has been removed

**Review by neallred — APPROVED** [review]
LGTM; I like how thorough the code elimination is!

**Pull request merged**